### PR TITLE
[Emacs Lisp] First draft of rna concepts

### LINKED ISSUE #531: [emacs-lisp] Extract Concepts from v2 exercise: rna-transcription
Extract the list of concepts required to solve the v2 emacs-lisp exercise: `rna-transcription`. For each concept, describe why it is required to solve the exercise. Keep in mind that an exercise can have multiple approaches (you can find community solutions here: `https://exercism.io/tracks/emacs-lisp/exercises/rna-transcription/solutions`). The goal of this task is to help determine the total list of concepts that make up `rna-transcription`.

The list of concepts should be stored in the following document: `languages/emacs-lisp/reference/exercise-concepts/rna-transcription.md`. [Use this link to create a PR to add this document][pr-link].

[pr-link]: ../new/master?filename=languages/emacs-lisp/reference/exercise-concepts/rna-transcription.md

### PR DESCRIPTION
**Pull request by maurelio1234**
fixes #531

### CODE REVIEW

**Comment by maurelio1234** [comment]
For now I just looked at the existing solutions, still need to organize the document a little bit

**Comment by maurelio1234** [comment]
Related: https://github.com/exercism/v3/pull/539

**Comment by maurelio1234** [comment]
There goes a V0... tell me what you think about it

**Comment by ErikSchierboom** [comment]
@maurelio1234 Could you point me to the solution you've used to extract the Concepts from?

**Comment by maurelio1234** [comment]
I looked at many different community solutions trying to find different approaches

https://exercism.io/tracks/emacs-lisp/exercises/rna-transcription/solutions

File: languages/emacs-lisp/reference/exercise-concepts/rna-transcription.md [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+
+- lists: a string is a list
+- calling other functions: usually one defines new functions
+- mapconcat/mapcar/concat/dolist/: to process letters in a string
+- lambdas: people usually do the transcoding in a lamba on a seq-map
+
+# Style
+
+- cl-labels: defining functions inline
+- namespaces: or the lack of namespaces... how to name your functions properly
```
Comment (ErikSchierboom) [inline languages/emacs-lisp/reference/exercise-concepts/rna-transcription.md]: Is this a Concept, or does this not exist?

File: languages/emacs-lisp/reference/exercise-concepts/rna-transcription.md [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+- docstring: not required, but part of the discussion with students
+- comments: not required, but part of the discussion when they are off
+- naming standards in elisp
+
+# Control structures
+
+- if: to detect if the code is invalid
+- error: raising errors when invalid
+- condition-case: to handle each case
+- cond: to handle each case
```
Comment (ErikSchierboom) [inline languages/emacs-lisp/reference/exercise-concepts/rna-transcription.md]: Is the `cond` part of `condition-case`, or is it a separate Concept?

File: languages/emacs-lisp/reference/exercise-concepts/rna-transcription.md (hunk 1 above)
Comment (maurelio1234) [reply languages/emacs-lisp/reference/exercise-concepts/rna-transcription.md]: Maybe it should be `naming conventions`

Comment (maurelio1234) [reply languages/emacs-lisp/reference/exercise-concepts/rna-transcription.md]: I changed the text

File: languages/emacs-lisp/reference/exercise-concepts/rna-transcription.md (hunk 2 above)
Comment (maurelio1234) [reply languages/emacs-lisp/reference/exercise-concepts/rna-transcription.md]: hmmm, it's more a kind of `try/catch`, I will merge it with `error`, it should probably be presented together

**Pull request merged**